compiler/basicTypes/Id.hs

```diff
@@ -289,35 +290,35 @@ mkVanillaGlobalWithInfo = mkGlobalId VanillaId
 
 
 -- | For an explanation of global vs. local 'Id's, see "Var#globalvslocal"
-mkLocalId :: Name -> Mult -> Type -> Id
-mkLocalId name w ty = mkLocalIdWithInfo name w ty vanillaIdInfo
+mkLocalId :: Name -> Mult -> UsageAnnotation -> Type -> Id
+mkLocalId name w ue ty = mkLocalIdWithInfo name w ue ty vanillaIdInfo
  -- It's tempting to ASSERT( not (isCoVarType ty) ), but don't. Sometimes,
  -- the type is a panic. (Search invented_id)
 
 -- | Make a local CoVar
 mkLocalCoVar :: Name -> Type -> CoVar
 mkLocalCoVar name ty
   = ASSERT( isCoVarType ty )
-    Var.mkLocalVar CoVarId name Omega ty vanillaIdInfo
+    Var.mkLocalVar CoVarId name Omega zeroUA ty vanillaIdInfo
 
 -- | Like 'mkLocalId', but checks the type to see if it should make a covar
-mkLocalIdOrCoVar :: Name -> Mult -> Type -> Id
-mkLocalIdOrCoVar name w ty
-  | isCoVarType ty = ASSERT(eqType w Omega) mkLocalCoVar name   ty
-  | otherwise      = mkLocalId    name w ty
+mkLocalIdOrCoVar :: Name -> Mult -> UsageAnnotation -> Type -> Id
```

Comment:
@monoidal can you look this up as part of the refactoring you want to perform?

It will probably make the diff bigger rather than smaller, but on the other hand, this may get us some of the way of fully classifying new identifiers as let-bound vs lambda-or-case-bound.